src/Parallel/Command/WorkerCommandLineFactory.php

```diff
@@ -173,8 +173,13 @@ private function mirrorCommandOptions(InputInterface $input, array $mainCommandO
                 continue;
             }
 
-            $workerCommandOptions[] = self::OPTION_DASHES . $mainCommandOptionName;
-            $workerCommandOptions[] = escapeshellarg($optionValue);
+            if ($mainCommandOptionName === 'memory-limit') {
+                // symfony/console does not accept -1 as value without assign
+                $workerCommandOptions[] = '--' . $mainCommandOptionName . '=' . $optionValue;
```

Comment:
does this line miss escaping?

Reply:
I've tried this line with shellescapearg and it would work. However, since the option name was not originally shellescaped, I did not use it for this special case and the option value is already validated by the main command and can therefore only accept valid values.
But please let me know if you think I should use shellescape in this case and I'll apply it

Reply:
IMO we should escape `optionValue` for symmetri with the `else` case

Reply:
Ok, that is a good point. I've pushed another commit taking care of escaping the shell option value